flask/flask-launcher/src/main/java/net/corda/flask/launcher/JarCache.java

```diff
@@ -114,28 +117,30 @@ public JarCache(String appName) {
     public Map<String, Path> extract(Manifest manifest) {
         byte[] buffer = new byte[Flask.Constants.BUFFER_SIZE];
         ClassLoader cl = Launcher.class.getClassLoader();
-        for (Map.Entry<String, Attributes> entry : manifest.getEntries().entrySet()) {
-            String jarEntryName = entry.getKey();
-            Attributes attributes = entry.getValue();
-            if (jarEntryName.startsWith(Flask.Constants.LIBRARIES_FOLDER + '/') && attributes.getValue(Flask.ManifestAttributes.ENTRY_HASH) != null) {
-                String jarName = jarEntryName.substring(jarEntryName.lastIndexOf('/') + 1);
-                String hash = Flask.bytes2Hex(Base64.getDecoder().decode(attributes.getValue(Flask.ManifestAttributes.ENTRY_HASH)));
-                Path destination = libDir.resolve(hash).resolve(jarName);
-                extractedLibraries.put(hash, destination);
-                if (!Files.exists(destination)) {
-                    Files.createDirectories(destination.getParent());
-                    InputStream is = cl.getResourceAsStream(jarEntryName);
-                    log.debug("Extracting '{}' to '{}'", jarEntryName, destination);
-                    if (is != null) {
-                        try {
-                            try (OutputStream os = new BufferedOutputStream(Files.newOutputStream(destination))) {
-                                Flask.write2Stream(is, os, buffer);
+        try (LockFile lf = LockFile.acquire(extractionLock, false)) {
+            for (Map.Entry<String, Attributes> entry : manifest.getEntries().entrySet()) {
+                String jarEntryName = entry.getKey();
+                Attributes attributes = entry.getValue();
+                if (jarEntryName.startsWith(Flask.Constants.LIBRARIES_FOLDER + '/') && attributes.getValue(Flask.ManifestAttributes.ENTRY_HASH) != null) {
+                    String jarName = jarEntryName.substring(jarEntryName.lastIndexOf('/') + 1);
+                    String hash = Flask.bytes2Hex(Base64.getDecoder().decode(attributes.getValue(Flask.ManifestAttributes.ENTRY_HASH)));
+                    Path destination = libDir.resolve(hash).resolve(jarName);
+                    extractedLibraries.put(hash, destination);
+                    if (!Files.exists(destination)) {
+                        Files.createDirectories(destination.getParent());
+                        InputStream is = cl.getResourceAsStream(jarEntryName);
+                        log.debug("Extracting '{}' to '{}'", jarEntryName, destination);
+                        if (is != null) {
+                            try {
+                                try (OutputStream os = new BufferedOutputStream(Files.newOutputStream(destination))) {
+                                    Flask.write2Stream(is, os, buffer);
+                                }
```

Comment:
Do we need to use a `BufferedOutputStream` here? We already have a `buffer[]`.

Reply:
correct, we don't need it!